elasticsearch-api/utils/thor/generate_api.rb

```diff
@@ -35,22 +35,22 @@ module Utils
       #
       def __parse_java_source(path)
         path  += '/' unless path =~ /\/$/ # Add trailing slash if missing
-        prefix = "src/main/java/org/elasticsearch/rest/action"
+        prefix = "core/src/main/java/org/elasticsearch/rest/action"
 
-        java_rest_files = Dir["#{path}#{prefix}/**/*.java"]
+        java_rest_files = Dir["#{path}#{prefix}/**/Rest*Action.java"]
 
         map = {}
 
         java_rest_files.sort.each do |file|
           content = File.read(file)
           parts   = file.gsub(path+prefix, '').split('/')
-          name    = parts[0, parts.size-1].reject { |p| p =~ /^\s*$/ }.join('.')
+          name    = parts[-1].scan(/Rest(.+?)[A-Z]*Action\.java/).first.first.downcase + '_' + parts[-2]
```

Comment:
@karmi I see. I just spent some time thinking if there are any good patterns in how developers are naming JSON files, and it seem like the rules are a bit value. One thing for sure – naming JSON files can't be directlyt inferred from how java rest action files are named.

Given your response, I'm starting to think there's no use for generating JSON anymore. So maybe it's worth to remove this part entirely, and always rely on JSON files in rest-api-spec? JSON specs in that folder would be a considered as single source of truth for when generating the API code.

> So maybe it's worth to remove this part entirely

What do you think?